tasks/broad/RNAWithUMIsTasks.wdl

```diff
@@ -614,3 +614,304 @@ task MarkDuplicatesUMIAware {
     }
 }
 
+
+task formatPipelineOutputs {
+  input {
+    String output_basename
+    String transcriptome_bam
+    String transcriptome_bam_index
+    String transcriptome_duplicate_metrics
+    String output_bam
+    String output_bam_index
+    String duplicate_metrics
+    String rnaseqc2_gene_tpm
+    String rnaseqc2_gene_counts
+    String rnaseqc2_exon_counts
+    String rnaseqc2_fragment_size_histogram
+    String rnaseqc2_metrics
+    String picard_rna_metrics
+    String picard_alignment_summary_metrics
+    String picard_insert_size_metrics
+    String picard_insert_size_histogram
+    String picard_base_distribution_by_cycle_metrics
+    String picard_base_distribution_by_cycle_pdf
+    String picard_quality_by_cycle_metrics
+    String picard_quality_by_cycle_pdf
+    String picard_quality_distribution_metrics
+    String picard_quality_distribution_pdf
+    String? picard_fingerprint_summary_metrics
+    String? picard_fingerprint_detail_metrics
+    File unified_metrics
+  }
+
+  String outputs_json_file_name = "outputs_to_TDR_~{output_basename}.json"
+
+  command <<<
```

Comment:
Could we move this python block out to live in the `broadinstitute/horsefish:emerge_scripts` docker image instead

Reply:
I'd prefer it live here because this is essentially a config file (mapping the pipeline outputs to the TDR dataset). But can move it if you feel strongly - what do you think?

Reply:
Eh I wouldn't say I feel _strongly_, just more a housekeeping thing!

@jessicaway @gbggrant care to weigh in and tell me if I'm being nit picky?

Reply:
Totally fair - I was going off of what Sushma did for the Arrays wrapper wdl (https://github.com/broadinstitute/warp/blob/emerge_wdl/pipelines/broad/arrays/single_sample/arrays_to_TDR.wdl#L110) - but I will happily go with the team's consensus here.

Reply:
I don't have a strong feeling on this. Is broadinstitute/horsefish:emerge_scripts a standard place for python scripts? I don't think putting *all* the python into a standardized docker makes sense. Might want to group them by function?  That is, should there be a broadinstitute/horsefish:tcap_scripts docker?

Reply:
Oh, I see broadinstitute/horsefish:emerge_scripts is the docker you are using for this task's runtime (I had thought it was a central standardized location).

Reply:
right, it's definiitely not central or standardized. i was planning to make a horsefish:tcap_scripts docker for the other big chunk of python.

Reply:
i don't think having different image tags for different scripts would be easy to track, i think it would be better to have distinct _images_, instead of tags

Reply:
this is honestly out of scope but we do really need to come up with a better way to standardize our containers around all these one off python scripts...but for this instance it may make sense to either

1. just add this script into that horsefish container (what is it anyway?)
2. build our own image just for this script that we can manage within warp like we do with our other pipelines

Reply:
yeah. horsefish is Field Engineering's repo, sort of a mish-mash of tools. on the one hand this makes sense to me to have there because this should be a temporary solution (i.e. these tasks should at some point be replaceable by product features in TDR/WFL). on the other hand, i like it in warp so that everything is together in one repo.

i'd vote 2, but again will defer to you all on how you think this should be organized.

Reply:
If this was something that required a more rigorous building process (building C binaries, adding weird external libraries etc.) we would probably choose 2 but since this is just a python script I think we can just go with option 1 for now

I'll add a ticket on our board to circle back to this and do a comb through of whats in horsefish and if we want to pick out the salient pieces from it and merge it with all of our other scripts for skylab etc.

Reply:
ok sounds great. thanks for reviewing and guiding towards a responsible decision! will work on building the new docker this afternoon

Reply:
ok the large script is now in the docker image. this thread got a little sidetracked - how do we feel about keeping this section in the WDL, since its primary function is reconfiguring the outputs into a file for TDR ingest?